src/listpack.c

```diff
@@ -1305,6 +1318,15 @@ unsigned char *lpReplaceInteger(unsigned char *lp, unsigned char **p, long long
     return lpInsertInteger(lp, lval, *p, LP_REPLACE, p);
 }
 
+int lpReplaceIntegerSameLen(unsigned char *lp, unsigned char *p, long long lval) {
```

Comment:
I don't like this method because it is only prepared for the stream. If the lengths are the same, actually we won't move the memory?

Reply:
Yes, we won't realloc the listpack if the replaced item is the same length as the original one. 
If we have to change the length of the flags integer and realloc the listpack, we give up updating the flag and perform a garbage collection to physically delete the entries.